fmriprep/interfaces/freesurfer.py

```diff
@@ -246,3 +285,70 @@ def detect_inputs(t1w_list, t2w_list=None, hires_enabled=True):
     # https://surfer.nmr.mgh.harvard.edu/fswiki/SubmillimeterRecon
     mris_inflate = '-n 50' if hires else None
     return (t2w, hires, mris_inflate)
+
+
+def refine_aseg(aseg, ball_size=4):
+    """
+    First step to reconcile ANTs' and FreeSurfer's brain masks.
+
+    Here, the ``aseg.mgz`` mask from FreeSurfer is refined in two
+    steps, using binary morphological operations:
+
+      1. With a binary closing operation the sulci are included
+         into the mask. This results in a smoother brain mask
+         that does not exclude deep, wide sulci.
+
+      2. Fill any holes (typically, there could be a hole next to
+         the pineal gland and the corpora quadrigemina if the great
+         cerebral brain is segmented out).
+
+
+    """
+    # Read aseg data
+    bmask = aseg.copy()
+    bmask[bmask > 0] = 1
+    bmask = bmask.astype(np.uint8)
+
+    # Morphological operations
+    selem = sim.ball(ball_size)
+    newmask = sim.binary_closing(bmask, selem)
+    newmask = binary_fill_holes(newmask.astype(np.uint8), selem).astype(np.uint8)
+
+    return newmask.astype(np.uint8)
+
+
+def grow_mask(anat, aseg, ants_segs, ww=7, zval=2.0, bw=4):
+    """
+    Grow mask including pixels that have a high likelihood.
+    GM tissue parameters are sampled in image patches of ``ww`` size.
+
+    This is inspired on mindboggle's solution to the problem:
+    https://github.com/nipy/mindboggle/blob/master/mindboggle/guts/segment.py#L1660
+
+    """
+    selem = sim.ball(bw)
+
+    aseg[aseg == 42] = 3  # Collapse both hemispheres
+    gm = anat.copy()
+    gm[aseg != 3] = 0
+
+    refined = refine_aseg(aseg)
+    newrefmask = sim.binary_dilation(refined, selem) - refined
+    indices = np.argwhere(newrefmask > 0)
+    for pixel in indices:
+        # When ATROPOS identified the pixel as GM, set and carry on
+        if ants_segs[tuple(pixel)] == 2:
```

Comment:
These lines incorporate Arno's concepts for relabeling pixels excluded by FreeSurfer. However, my impression is that this adds a lot of pixels wrongly identified as GM by ATROPOS during `antsBrainExtraction.sh` (the segmentation comes from there).

Let me know if we want to keep it or discard it.

Reply:
It would be useful to see an example of this in action (in other words which voxels are wrongly relabeled).

Reply:
Okay, I'll get some examples.

I have to say that Arno's segmentations are probably a lot better, since he gets them from `antsCorticalThickness.sh`. The idea is probably great, but currently we can't apply it properly.

Reply:
https://github.com/oesteban/notebooks/blob/f5cf852d212867b32e7bf38bda2424e7cc132a35/FMRIPREP-issue431.ipynb - discussion moved to the main thread.